app/src/main/java/pm/gnosis/heimdall/ui/safe/details/transactions/SafeTransactionsViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package pm.gnosis.heimdall.ui.safe.details.transactions
+
+import io.reactivex.Observable
+import pm.gnosis.heimdall.common.utils.Result
+import pm.gnosis.heimdall.common.utils.mapToResult
+import pm.gnosis.heimdall.common.utils.onErrorDefaultBeforeThrow
+import pm.gnosis.heimdall.data.repositories.GnosisSafeRepository
+import pm.gnosis.heimdall.ui.base.Adapter
+import pm.gnosis.heimdall.utils.mapScanToMappedResult
+import java.math.BigInteger
+import javax.inject.Inject
+
+
+class SafeTransactionsViewModel @Inject constructor(
+        private val safeRepository: GnosisSafeRepository
+) : SafeTransactionsContract() {
+
+    companion object {
+        const val PAGE_SIZE = 20
+    }
+
+    private var loadingMore: Boolean = false
+    private var cachedResults: IndexedResults? = null
+    private var address: BigInteger? = null
```

Comment:
Can this address be null at any point of the view model Lifecycle? Maybe a `lateinit` would be better.

Reply:
with a lateinit I cannot do the check if it has changed on setup (would throw an uninitialized exception)

I could catch that, but that would be quite ugly